applications/tari_console_wallet/src/ui/state/wallet_event_monitor.rs

```diff
@@ -113,6 +121,22 @@ impl WalletEventMonitor {
                         trace!(target: LOG_TARGET, "Wallet Event Monitor received wallet connectivity status changed");
                         self.trigger_peer_state_refresh().await;
                     },
+                    Ok(_) = software_update_notif.changed() => {
+                        trace!(target: LOG_TARGET, "Wallet Event Monitor received wallet auto update status changed");
+                        let update:Option<SoftwareUpdate> = match *software_update_notif.borrow() {
+                            Some(ref update) => Some(update.clone()),
+                            None => None
+                        };
```

Comment:
📎 This this would work
```suggestion
                        let update = software_update_notif.borrow().as_deref().cloned();
```

Reply:
That throws : **error[E0623]: lifetime mismatch**
This part of code was giving me headache with error "future is not `Send` as this value is used across an await". I didn't find any nice code. But I do want to know how to do it simple. Because it's quite ugly now...

Reply:
That throws : **error[E0599]: the method `as_deref` exists for struct `tokio::sync::watch::Ref<'_, Option<tari_p2p::auto_update::SoftwareUpdate>>`, but its trait bounds were not satisfied**
This part of code was giving me headache with error "future is not Send as this value is used across an await". I didn't find any nice code. But I do want to know how to do it simple. Because it's quite ugly now...

Reply:
Fixed, Thanks for the via chat.